Examples/Liquidate/BorrowSystem.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: agpl-3.0
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+interface IERC20 {
+	function transfer(address to, uint value) external;
+	function transferFrom(address from, address to, uint value) external;
+}
+
+interface IOracle {
+	function get(IERC20 token1, IERC20 token2) external returns (uint256);
+}
+
+contract SimpleBorrowSystem {
+	// oracle to be used
+	IOracle oracle;
+	// ERC20 collateralToken
+	IERC20 collateralToken;
+	// ERC20 borrowToken 
+	IERC20 borrowToken;
+
+	// amount of borrowToken user borrowed
+	mapping(address => uint256) public userBorrowAmount;
+	// amount of collateralToken user has deposited
+	mapping(address => uint256) public userCollateralAmount;
+
+	// actions for the batchCalls
+	uint8 internal constant CALL_BORROW = 1;
+	uint8 internal constant CALL_REPAY = 2;
+	uint8 internal constant CALL_LIQUIDATE = 3;
+
+	/* Borrow amount and provide collateral */
+	function borrow(uint256 borrowAmount, uint256 collateralAmount) public {
```

Comment:
The user can choose how much they can put as collateral? The idea is if they try to cheat, they are insolvent and someone would liquidate them?

Reply:
the can put collateral and borrow as much as they want as long that they are in solvet state. The bug is in liquidation when calling from batch